src/helpers.ts

```diff
@@ -48,18 +49,14 @@ export function unescape(html) {
 
 const caret = /(^|[^\[])\^/g;
 
-/**
- * @param {string | RegExp} regex
- * @param {string} opt
- */
-export function edit(regex, opt) {
+export function edit(regex: Rule, opt?: string) {
   regex = typeof regex === 'string' ? regex : regex.source;
   opt = opt || '';
   const obj = {
-    replace: (name, val) => {
+    replace: (name: string | RegExp, val: any) => {
```

Comment:
Can we have a more specific type than `any`?

Reply:
Sure. Adjusted to `string | RegExp`, but had to adapt the code below to handle it in a type safe way